lib/ht_members.rb

```diff
@@ -1,17 +1,151 @@
 # frozen_string_literal: true
 
-# WIP: cache of information about HathiTrust members
+require "mysql2"
+require "dotenv"
+
+Dotenv.load(".env")
+
+=begin
+
+Cache of information about HathiTrust members
+The data may be mocked if you pass it in to new(),
+otherwise it will try to read it from db, and if that fails
+it will use canned data (CANNED_DATA).
+
+@mocked and @canned are set accordingly and readable, so you can
+tell where the data is coming from
+
+Usage:
+
+  htm = HTMembers.new()
+  puts "using canned data? #{htm.canned}"
+  cc = htm.get("yale")["country_code"]
+  wt = htm.get("harvard")["weight"]
+
+Or if you want to provide your own mock data:
+
+  htm = HTMembers.new({
+    "haverford" => {
+      "country_code" => "us",
+      "weight"       => 0.67,
+    }
+  })
+  puts "using canned data? #{htm.canned}"
+  puts "using mocked data? #{htm.mocked}"
+  htm.get("haverford")["country_code"]
+  htm.get("haverford")["weight"]
+
+=end
+
 class HTMembers
+  def self.template(cc, wt, sym)
+    {
+      "country_code" => cc,
+      "weight"       => wt,
+      "oclc_sym"     => sym,
+    }
+  end
 
-  def initialize(*members)
-    @members = members.empty? ? load_from_db : members
+  # If we get more cases like universityofcalifornia then this might
+  # look better as a series of SELECT ... UNION SELECT ....
+  SQL = %w<
```

Comment:
The more I think about it I don't think we need the distinct/mapto query. In particular, we aren't using this to check if a holdings ID is valid as such or that we expect to get it, and so long as we load the holdings under the correct ID I don't think we need to worry about filtering or mapping the institution IDs here. If we need that functionality for billing calculation we can use it.